Tests/NIOPosixTests/DatagramChannelTests.swift

```diff
@@ -698,29 +698,39 @@ final class DatagramChannelTests: XCTestCase {
                 XCTAssertNoThrow(try sendChannel.close().wait())
             }
             
-            var buffer = sendChannel.allocator.buffer(capacity: 1)
-            buffer.writeRepeatingByte(0, count: 1)
             let ecnStates: [NIOExplicitCongestionNotificationState] = [.transportNotCapable,
                                                                        .congestionExperienced,
                                                                        .transportCapableFlag0,
                                                                        .transportCapableFlag1]
-            for ecnState in ecnStates {
-                let writeData = AddressedEnvelope(remoteAddress: receiveChannel.localAddress!,
-                                                  data: buffer,
-                                                  metadata: .init(ecnState: ecnState, packetInfo: expectedPacketInfo))
+            for (ecnStateIdx, ecnState) in ecnStates.enumerated() {
+                // Datagrams may be received out-of-order, so we use the payload to associate the read.
+                let writeData = AddressedEnvelope(
+                    remoteAddress: receiveChannel.localAddress!,
+                    data: sendChannel.allocator.buffer(
+                        integer: ecnStateIdx,
+                        endianness: .big,
+                        as: Int.self
```

Comment:
There should be no need for `endianness` or `as` here: the type inference should solve the latter and the default should solve the former.

Reply:
Was just being paranoid so that I was guaranteeing the same format on the read- and write-paths.

Simplified in 797805d27.